pkg/services/payment_request/payment_request_creator.go

```
@@ -16,11 +17,14 @@ func NewPaymentRequestCreator(db *pop.Connection) services.PaymentRequestCreator
 	return &paymentRequestCreator{db}
 }
 
-func (p *paymentRequestCreator) CreatePaymentRequest(paymentRequest *models.PaymentRequest) (*models.PaymentRequest, *validate.Errors, error) {
+func (p *paymentRequestCreator) CreatePaymentRequest(paymentRequest *models.PaymentRequest) (*models.PaymentRequest, error) {
 	verrs, err := p.db.ValidateAndCreate(paymentRequest)
-	if err != nil || verrs.HasAny() {
-		return nil, verrs, err
+	if err != nil {
+		return nil, fmt.Errorf("failure creating payment request: %w", verrs)
+	}
+	if verrs.HasAny() {
+		return nil, fmt.Errorf("validation error saving PaymentRequest: %w",  verrs)
 	}
 
-	return paymentRequest, verrs, err
+	return paymentRequest, err
```

It does seem that we're assuming here that we will have either `verrs` or `err`, and not possibly both.  If we could have both, and still want to just return a single error, we may need to combine the two error strings together and create a new error and then wrap that so we don't lose any info.  I'm not sure if I have a strong preference yet about what makes the most sense.  Some of this came up recently: https://ustcdp3.slack.com/archives/CP6F568DC/p1574180834227000

---

Thanks for the link.  I'm a little confused by the conclusion in the thread.  Sounds like Jacquie is saying we should check `verrs` first if we are checking if either are true, but in her code she is checking `err` first, unless I'm misunderstanding. `if err != nil || verrs.HasAny() {. . .}`

---

I think in the code above, she's looking for either an `err` or some `verrs` and then building up a single error string (from either or both) and making a new error out of that, then wrapping it.  Given our current model validation code, I don't think it's likely we'll have both `verrs` and an `err` on the same execution, though (although theoretically possible given the validation API).  The key thing is that we stop execution and report the error to the caller, which you're doing.  What's not as clear to me is if we should be returning separate `verrs` (unwrapped) and `err` or combining them into one if we control the method signature -- I see a few `creator` service objects and the query builder that appear to do the former.